google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/DdlBatchTest.java

```diff
@@ -0,0 +1,372 @@
+    batch.getReadTimestamp();
+  }
+
+  @Test
+  public void testWrite() {
+    DdlBatch batch = createSubject();
+    exception.expect(SpannerExceptionMatcher.matchCode(ErrorCode.FAILED_PRECONDITION));
+    batch.write(Mutation.newInsertBuilder("foo").build());
+  }
+
+  @Test
+  public void testWriteIterable() {
+    DdlBatch batch = createSubject();
+    exception.expect(SpannerExceptionMatcher.matchCode(ErrorCode.FAILED_PRECONDITION));
+    batch.write(Arrays.asList(Mutation.newInsertBuilder("foo").build()));
+  }
+
+  @Test
+  public void testIsReadOnly() {
+    DdlBatch batch = createSubject();
+    assertThat(batch.isReadOnly(), is(false));
+  }
+
+  @Test
+  public void testGetStateAndIsActive() {
+    DdlBatch batch = createSubject();
+    assertThat(batch.getState(), is(UnitOfWorkState.STARTED));
+    assertThat(batch.isActive(), is(true));
+    batch.runBatch();
+    assertThat(batch.getState(), is(UnitOfWorkState.RAN));
+    assertThat(batch.isActive(), is(false));
+
+    batch = createSubject();
+    assertThat(batch.getState(), is(UnitOfWorkState.STARTED));
+    assertThat(batch.isActive(), is(true));
+    batch.abortBatch();
+    assertThat(batch.getState(), is(UnitOfWorkState.ABORTED));
+    assertThat(batch.isActive(), is(false));
+
+    DdlClient client = mock(DdlClient.class);
+    doThrow(SpannerException.class).when(client).executeDdl(anyListOf(String.class));
+    batch = createSubject(client);
+    assertThat(batch.getState(), is(UnitOfWorkState.STARTED));
+    assertThat(batch.isActive(), is(true));
+    ParsedStatement statement = mock(ParsedStatement.class);
+    when(statement.getStatement()).thenReturn(Statement.of("CREATE TABLE FOO"));
+    when(statement.getSqlWithoutComments()).thenReturn("CREATE TABLE FOO");
+    when(statement.getType()).thenReturn(StatementType.DDL);
+    batch.executeDdl(statement);
+    boolean exception = false;
+    try {
+      batch.runBatch();
+    } catch (SpannerException e) {
+      exception = true;
+    }
+    assertThat(exception, is(true));
+    assertThat(batch.getState(), is(UnitOfWorkState.RUN_FAILED));
+    assertThat(batch.isActive(), is(false));
+  }
+
+  private static IsListOfStringsWithSize isEmptyListOfStrings() {
+    return new IsListOfStringsWithSize(0);
+  }
+
+  private static IsListOfStringsWithSize isListOfStringsWithSize(int size) {
+    return new IsListOfStringsWithSize(size);
+  }
+
+  private static class IsListOfStringsWithSize extends ArgumentMatcher<List<String>> {
+    private final int size;
+
+    private IsListOfStringsWithSize(int size) {
+      this.size = size;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public boolean matches(Object list) {
+      return ((List<String>) list).size() == size;
+    }
+  }
+
+  @Test
+  public void testRunBatch() {
+    DdlClient client = createDefaultMockDdlClient();
+    DdlBatch batch = createSubject(client);
+    batch.runBatch();
```

Comment:
following my previous comment, should we add assertions of the return type for `runBatch()`?

Reply:
Added two additional test cases to check for successful and failed batches and the actual return values.